### PR TITLE
update arrow-rs deps to latest master

### PR DESCRIPTION
**Pull request by alamb**
Update to the latest arrow-rs master along with a script to automate the process

 # Rationale for this change
Get the latest and greatest Arrow, and add a script to make it easier to do in the future

# What changes are included in this PR?
1. Update to latest arrow + code 
2. Script to automatically update dependencies in the future

# Example

```shell
python update_arrow_deps.py
Updating files in /Users/alamb/Software/arrow-datafusion to use sha ed00e4d4a160cd5182bfafb81fee2240ec005014
updating /Users/alamb/Software/arrow-datafusion/Cargo.toml
updating /Users/alamb/Software/arrow-datafusion/ballista/rust/core/Cargo.toml
updating /Users/alamb/Software/arrow-datafusion/ballista/rust/scheduler/Cargo.toml
updating /Users/alamb/Software/arrow-datafusion/ballista/rust/executor/Cargo.toml
updating /Users/alamb/Software/arrow-datafusion/ballista/rust/client/Cargo.toml
updating /Users/alamb/Software/arrow-datafusion/datafusion/Cargo.toml
updating /Users/alamb/Software/arrow-datafusion/datafusion-examples/Cargo.toml
updating /Users/alamb/Software/arrow-datafusion/benchmarks/Cargo.toml

```

### CODE REVIEW

File: dev/update_arrow_deps.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
```
Comment (alamb) [inline dev/update_arrow_deps.py:1]: I am sorry if this causes 👁️ 🩸

**Review by jorgecarleitao — APPROVED** [review]
LGTM. Thanks a lot for this.

**Comment by Dandandan** [comment]
Looks good 💯. I am wondering what the process for doing releases to crates.io is going to be? In that case we need to use release versions.

**Comment by alamb** [comment]
@Dandandan  -- I would like to get to the point where we do much more frequent arrow releases (like maybe weekly) and then datafusion will just pick them up automatically

I was imagining, for example, releasing  `4.1.0` shortly with these fixes. 

But I don't really have a plan / sorted out a process for doing this yet -- @andygrove  have you given the matter much thought?

**Comment by andygrove** [comment]
@alamb I agree that we want frequent releases. I have not had much time to think about what we do next. We should collaborate on a process in the google doc we have going for the migration as next step?

**Comment by alamb** [comment]
> We should collaborate on a process in the google doc we have going for the migration as next step?

I agree @andygrove  if you haven't had a chance, when I next get some time I will do so

**Pull request merged**